### PR TITLE
Instructions for migrating to custom theme folder

### PR DESCRIPTION
**Pull request by rolfsmeds**
Added instructions for refactoring from @JsModule and @CssImport based styling to the new theme folder format.

### CODE REVIEW

File: articles/v14-23-upgrade/final-steps/index.adoc [hunk 1]
```diff
@@ -74,3 +74,129 @@ include::instructions/_typescript.adoc[]
...
++
+----
+/* Before */
+@CssImport(value="text-field-styles.css", themeFor="vaadin-text-field")
+
+/* After */
+frontend/themes/my-theme/components/vaadin-text-field.css
+----
+
+6. Move other stylesheets next to `styles.css`
```
Comment (tomivirkki) [inline articles/v14-23-upgrade/final-steps/index.adoc:167]: Should we also advise to move [styles](https://github.com/vaadin/bakery-app-starter-flow-spring/blob/master/frontend/styles/shared-styles.js#L319) defined with `<custom-style>` element to this file?

File: articles/v14-23-upgrade/final-steps/index.adoc [hunk 2]
```diff
@@ -74,3 +74,129 @@ include::instructions/_typescript.adoc[]
...
++
+----
+frontend
+└── themes
+    └── my-theme
+        ├── components/
+        └── styles.css
+----
+
+5. Move all Vaadin component stylesheets, that have a `themeFor` parameter (or, if refactoring from JsModules, a `theme-for` attribute in the `dom-module`), into the `components` folder and rename them to correspond to the `themeFor` value (the component's HTML tagname)
```
Comment (dougjewsbury) [inline articles/v14-23-upgrade/final-steps/index.adoc:157]:
```suggestion
5. Move all Vaadin component stylesheets that have a `themeFor` parameter (or, if refactoring from JsModules, a `theme-for` attribute in the `dom-module`) into the `components` folder and rename them to correspond to the `themeFor` value (the component's HTML tag name).

```

File: articles/v14-23-upgrade/final-steps/index.adoc [hunk 3]
```diff
@@ -74,3 +74,129 @@ include::instructions/_typescript.adoc[]
...
+└── themes
+    └── my-theme
+        ├── components/
+        ├── styles.css
+        ├── navigation.css
+		├── dashboard.css
+        └── ...
+----
++
+and add import each one into `styles.css` using `@import`
```
Comment (dougjewsbury) [inline articles/v14-23-upgrade/final-steps/index.adoc:180]:
```suggestion
and import each one into `styles.css` using `@import`

```

**Comment by tarekoraby** [comment]
Eh .. Amplify deployment not working again 😞

File: articles/v14-23-upgrade/final-steps/index.adoc [hunk 4]
```diff
@@ -89,3 +89,152 @@ include::instructions/_typescript.adoc[]
...
 include::instructions/_iron.adoc[]
 ====
+
+
+== Replace JsModule Stylesheets With CssImports
+.Click and read if you have styles imported with `@JsModule` annotations.
+[%collapsible]
+====
+
+Loading stylesheets with the `@JsModule` annotation in Flow is deprecated. JS-wrapped stylesheets should be refactored to regular CSS files and loaded with @CssImport, or, preferably, using <<custom-theme-folder,the new theme folder format>> recommended for styling applications since Vaadin 19.
```
[the remark this answers is not in the document]
Comment (rolfsmeds) [reply articles/v14-23-upgrade/final-steps/index.adoc]: Done (even though I have no idea what good [annotationname] actually does 😅 )

**Comment by rolfsmeds** [comment]
@JsModule refac instructions now wrapped into expandable section now.

**Comment by tarekoraby** [comment]
The styling of JsModule in this line here looks weird.

image

**Comment by tarekoraby** [comment]
Also the content under "Refactor Styles to a Custom Theme Folder" should be made collapsible.

**Review by tarekoraby — APPROVED** [review]
We agreed in the daily to keep this section un-collapsible.

**Pull request merged**